go/mcap/indexed_message_iterator.go

```diff
@@ -144,9 +165,48 @@ func (it *indexedMessageIterator) parseSummarySection() error {
 			it.statistics = stats
 		case TokenFooter:
 			it.hasReadSummarySection = true
-			return nil
+			break lexerloop
+		}
+	}
+	it.chunksOverlap = false
+	switch it.indexHeap.order {
+	case FileOrder:
+		sort.Slice(it.chunkIndexesToLoad, func(i, j int) bool {
+			return it.chunkIndexesToLoad[i].ChunkStartOffset < it.chunkIndexesToLoad[j].ChunkStartOffset
+		})
```

Comment:
is this logic correct? It seems like it will not always result in "file order" records. Imagine for instance that I have,

[chunk [100 500]]
[chunk [50 300]]

And my query is for [50-500]. What should happen? It seems like this would reorder the chunks, whereas I would expect them both in file order.

Reply:
ChunkStartOffset is the byte offset of the chunk, not a message timestamp.

Reply:
ah, thanks. Seems fine.